src/item.cpp

```diff
@@ -5023,12 +5023,19 @@ units::mass item::weight( bool, bool integral ) const
 
 units::length item::length() const
 {
-    if( made_of( phase_id::LIQUID ) || is_soft() ) {
+    if( made_of( phase_id::LIQUID ) || ( is_soft() && is_container_empty() ) ) {
         return 0_mm;
     }
     if( is_corpse() ) {
         return units::default_length_from_volume<int>( corpse->volume );
     }
+    if( is_container() ) {
```

Comment:
this check is actually unecessary, since contents.all_items_top( item_pocket::pocket_type::CONTAINER ) already does some checks if there are any items in the container. the for loop would therefore not run at all if it didnt' have any contents in CONTAINER pockets.